### ISSUE TITLE
The parseFrontmatter option doesn't work with next-mdx-remote/rsc

### ISSUE THREAD

**Issue by jasongerbes** [hashicorp/next-mdx-remote#345]
## Expected behaviour

Using the `compileMDX` method of `next-mdx-remote/rsc` with `parseFrontmatter: true`, the expected behaviour is that any frontmatter should be parsed from the MDX source.

<img width="1033" alt="image" src="https://user-images.githubusercontent.com/13686180/221453786-4a94705a-6bc9-405c-bad9-621a5bc1e30d.png">

## Actual behaviour

The `parseFrontmatter` option has no affect on the `content` returned by `compileMDX`, and the `frontmatter` value is always `{}`.

<img width="1033" alt="image" src="https://user-images.githubusercontent.com/13686180/221453726-9bfd02f4-df04-4495-b379-00b3dbd62794.png">

## Reproduction steps

Minimal reproduction here: https://github.com/jasongerbes/next-mdx-remote-rsc

This reproduction uses the "Access Frontmatter outside of MDX" example from the README with a couple of fixes:

1. Added missing `await`
2. Removed extra "Hello World!" heading

```jsx
// app/page.tsx
import { compileMDX } from "next-mdx-remote/rsc";

export default async function Home() {
  const { content, frontmatter } = await compileMDX({
    source: `
      ---
      title: RSC Frontmatter Example
      ---
      This is from Server Components!
    `,
    options: { parseFrontmatter: true },
  });
  return (
    <>
      <h1>{frontmatter.title}</h1>
      {content}
    </>
  );
}

```

**Comment by chrisweb** [comment]
I can confirm your findings and I think I found the problem, there is space between the first line of frontmatter "---" and the opening of the template literal, if you change the example to this:

```tsx
  const { content, frontmatter } = await compileMDX({
    source: `---
      title: RSC Frontmatter Example
      ---
      This is from Server Components!`,
    options: { parseFrontmatter: true },
  });
```

then it should work

on top of the readme the example for the pages directory **getStaticProps** is correct the new ones are not

I got suspicious when I tried to load the content from a file and it actually worked with the same MDX file but when using the template literal it didn't anymore, so I knew it had something to do with its formatting

**Comment by jasongerbes** [comment]
@chrisweb you're right that it was an issue with whitespace. However, you also need to remove the whitespace before the other lines of the source:

```js
  const { content, frontmatter } = await compileMDX({
    source: `---
title: RSC Frontmatter Example
---
This is from Server Components!`,
    options: { parseFrontmatter: true },
  });
```
